### PR TITLE
[main < c-api-docs] Add missing C-API docs

### PR DESCRIPTION
**Pull request by imilinovic**
### Description

Add docs for all undocumented C-API changes made by mage-interns in the summer of 2023.
Everything here is already in the released build.

### Pull request type

Please check what kind of PR this is:

- [X] Fix or improvement of an existing page

### Related PRs and issues
[PR1221](https://github.com/memgraph/memgraph/pull/1221)
[PR1217](https://github.com/memgraph/memgraph/pull/1217)
[PR1103](https://github.com/memgraph/memgraph/pull/1103)

PR this doc page is related to: 
(especially necessary if the PR is related to a release)

Closes:
(paste the link to the issue it closes)

### Checklist:

- [x] Check all content with Grammarly
- [x] Perform a self-review of my code
- [x] Make corresponding changes to the rest of the documentation (consult with the DX team)
- [x] The build passes locally
- [x] My changes generate no new warnings or errors
- [x] Add a corresponding label
- [x] If release-related, add a product and version label
- [x] If release-related, add release note on product PR

### CODE REVIEW

**Review by vpavicic — APPROVED** [review]

**Comment by vpavicic** [comment]
I'll wait for a review from mpintaric :)

**Review by mpintaric55334 — APPROVED** [review]

**Pull request merged**
